### PR TITLE
BugFix : advanced Mode and camera state not persistent across Screen navigation

### LINKED ISSUE #296: CreateProfileScreen : Profile picture is not uploaded
### Problem 
selecting a profile picture when creating a profile is useless as it is not saved.

### Solution : 
```patch
   val onSignInClick = { createdUser: User ->
     userViewModel.authenticate(
         // On successful authentication
         { userId ->
           val user = createdUser.copy(public = createdUser.public.copy(userId = userId))
@@ -49,6 +50,7 @@ fun CreateProfileScreen(navigationActions: NavigationActions, userViewModel: Use
                   user,
                   {
                     userViewModel.setCurrentUser(user)
+                    userViewModel.updateUser(user, context)
                     Toast.makeText(context, validationToastText, Toast.LENGTH_SHORT).show()
                     navigationActions.navigateTo(TopLevelDestinations.MAP)
                   },
```

### PR DESCRIPTION
**Pull request by Pijukai**
Closes #296 #275 

## What is the Content of this PR ? 

This PR does the following : 

1. Addresses the bug that makes the camera State and the advanced mode not persistent during screen navigations
2. Add some Documentation 
3. Refactor `MapScreen` logic into `MapViewModel` to prevent persistency bugs. 

## How ? 

1. For the Advanced mode, simply refactoring the variable to the `MapViewModel` was enough to make it persistent during screen Navigation. For the Camera State , I updated the variable `cameraPosition` inside all the OnClick lambdas that are linked to a button leaving the screen. I also created the Boolean `cameraRecentering` in the  `MapViewModel` to enable the `LaunchedEffect` that recenters the camera on the puck only when the map is initialized the first time. Then it is disabled by the boolean. 
2. Added Documentation to the different variables initialized at the beginning of `MapScreen.kt` 
3. Alongside the variables that were refactored, i also refactored `DrawRectangles` and `DrawMarkers` and the enum `MapMode` that were inside `MapScreen` to `MapViewModel` 

## Code Coverage
image

I modified:

- ui.map (coverage 74% impossible to fix due to the same problem as previous PR) 
- model.map (coverage 82%)

## Additional Infos 

This PR also implements the fix of issue #296 since it is a one-liner solution.

### CODE REVIEW

File: app/src/main/java/com/github/se/cyrcle/model/map/MapViewModel.kt [hunk 1]
```diff
@@ -186,6 +232,72 @@ class MapViewModel : ViewModel() {
...
   }
 
+  /**
+   * Draw markers on the map.
+   *
+   * @param pointAnnotationManager the PointAnnotationManager to draw the markers on
+   * @param parkingList the list of parkings to draw
+   * @param bitmap the bitmap to use for the markers
+   */
+  fun drawMarkers(
```
[the remark this answers is not in the document]
Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/model/map/MapViewModel.kt:242]: If we're strictly MVVM here... a draw function takes data and puts it into the UI, which is exactly the purpose of a ViewModel in my opinion. It has its place here

File: app/src/main/java/com/github/se/cyrcle/model/map/MapViewModel.kt [hunk 2]
```diff
@@ -186,6 +232,72 @@ class MapViewModel : ViewModel() {
...
+   *
+   * @param polygonAnnotationManager the polygon annotation manager
+   * @param parkingsList the list of locations to draw
+   */
+  fun drawRectangles(
+      polygonAnnotationManager: PolygonAnnotationManager?,
+      plabelAnnotationManager: PointAnnotationManager?,
+      parkingsList: List<Parking>
+  ) {
+    val annotations: MutableList<PolygonAnnotationOptions> = mutableListOf()
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/model/map/MapViewModel.kt:269]: Some inline comments here would be nice, it's a little hard to follow

**Review by LouisJamesVasseur — COMMENTED** [review]
Don't really much to say, the UI improvements are noticeable and the code is clean! I'm experiencing the same issue as @matttsss though w/ the needing to double click, please fix that and I'll approve!

**Comment by Pijukai** [comment]
The bug should be fixed now. It came from the fact that now that the user mode is in view Model, the program runasynchronously and so inside the `clickedChecked` of the Switch in MapScreen the instructions don't run in the order they are supposed to.

File: app/src/main/java/com/github/se/cyrcle/model/map/MapViewModel.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pijukai) [reply app/src/main/java/com/github/se/cyrcle/model/map/MapViewModel.kt:242]: Since we are manipulating the `pointAnnotationManager` and the data itself for me it should belong in the view model. I don't think there is a problem since this way the view model can control what data is displayed on the map.

File: app/src/main/java/com/github/se/cyrcle/model/map/MapViewModel.kt [hunk 3]
```diff
@@ -186,6 +232,72 @@ class MapViewModel : ViewModel() {
...
+    }
+  }
+
+  /**
+   * Draw the rectangles on the map
+   *
+   * @param polygonAnnotationManager the polygon annotation manager
+   * @param parkingsList the list of locations to draw
+   */
+  fun drawRectangles(
```
[the remark this answers is not in the document]
Comment (Pijukai) [reply app/src/main/java/com/github/se/cyrcle/model/map/MapViewModel.kt:264]: I think that is the same answer then but don't hesitate to tell me if you want to discuss it.

File: app/src/main/java/com/github/se/cyrcle/model/map/MapViewModel.kt (hunk 2 above)
Comment (Pijukai) [reply app/src/main/java/com/github/se/cyrcle/model/map/MapViewModel.kt:271]: Done. Tell me if that works for you.

**Review by LouisJamesVasseur — APPROVED** [review]
Good job on the fix!
